c10/core/DispatchKeySet.cpp

```diff
@@ -28,6 +28,7 @@ constexpr DispatchKeySet non_functional_backend_dispatch_keyset =
     backend_dispatch_keyset
         // XLA and LazyTensor are currently the only 2 backends in core
         // that use functionalization pass in eager mode.
+        .remove(DispatchKey::Sparse)
```

Comment:
sorry for the late review. I'm not sure this'll do exactly what you want (but maybe we should change that).

This keyset maps to the `CompositeExplicitAutogradNonFunctional` key. Today, the only kernels that get registered to this key are kernels that:

(1) Are for functional ops
(2) Who's kernels internally call **view** ops

There are a bunch of functional ops with kernels that call mutating ops, but they don't get registered to that dispatch key today. This is mostly because XLA/LTC are actually perfectly capable of handling mutations on their own - so why bother preventing them from re-using composite kernels that they're perfectly capable of handling?

If the functional->mutating decomps are an important class of kernels to remove from sparse, then maybe we should add a separate alias key for it - or even just re-purpose this key (the main benefit to LTC/XLA is just around error checking).